### DISCUSSION TITLE
Separate Wallet/Plotting/Farming

### DISCUSSION THREAD

**Discussion by CesareDj** [Chia-Network/chia-blockchain-gui#2384] [Ideas]
Right now, in Settings -> General we can choose between two modes with a radio button:

If you choose wallet mode, you show hidden options for farming. If you check the farm option, this farming options becomes hidden.

This is ok, but, if we want only to plot, we must choose farming option and that update the DB, but, what happens if we have two different computers for different pourposes ? For example, one for plotting and other for farming. Well, we must have two times de DB, and until download it completly, it is a cost of CPU unnecesary in this cases.

So, I want to know if will be better to change from a radio buttons to checkboxes with three options. Wallet, and we could see "Tokens-NFTs-Credentials-Offers-Contacts"; Farming, with the rest of the actual options without "Add parcel" on this options; Plotting, if you check this option, you show "Add parcels" like we can see in the UI right now and other sepparate tab "Create" where whe can see the "Add a parcel" page and "Pools" to see then if you want.

The objetive to separate this is plotting without the need to download the DB.

I hope you find this idea of your interest.
César.

**Comment by CesareDj** [comment]
... or you could show allways "make a parcel".

**Comment by BrandtH22** [comment]
@CesareDj , note I transferred this enhancement request to the GUI repo for better visibility 👍

**Comment by OverActiveBladderSystem** [comment]
> Right now, in Settings -> General we can choose between two modes with a radio button:
> 
> So, I want to know if will be better to change from a radio buttons to checkboxes with three options. Wallet, and we could see "Tokens-NFTs-Credentials-Offers-Contacts"; Farming, with the rest of the actual options without "Add parcel" on this options; Plotting, if you check this option, you show "Add parcels" like we can see in the UI right now and other sepparate tab "Create" where whe can see the "Add a parcel" page and "Pools" to see then if you want.
> 
> The objetive to separate this is plotting without the need to download the DB.

Just to extend this idea a little based on listening to past experiences of windows users who have utilized multiple machines for plotting faster, as well as bypassing the setup of certificates for remote harvesters...

**Have the ability to turn each service on/off independent of the others, and/or choose a remote full node.**

I think if there was a GUI option to change the "Full Node" service from "Local IP" (the machine you're using) to "Remote IP" (another computer on/off your local network) that would accomplish the goal (while also allowing users to harvest/farm those plots on the machine as they are created with minimal setup and knowledge).

**Rather than run a remote Harvester only service, add the Farmer service to each machine to simplify the setup steps.**

Running both the Harvester/Farmer service on a machine thats plotting should be minimal impact, so if the user already has a Full Node service running on another machine, it may be easier for the GUI user to stop the local Full Node service and provide the remote Full Node IP/Port to utilize in the GUI (this also will eliminate the step of setting up certificates for remote harvesting, a real pain point for some GUI users).

Side note, this also may help some users who use separate "light farming" software for various reasons (bandwidth, compute, configuration, etc), now pools who offer this service to their users would be able to tell the user to change the local Full Node to use a remote Full Node by entering the IP/Port in the GUI rather than utilize a 3rd party program.

**Reply by CesareDj** [reply]
It does have a significant impact on the CPU at the time, as downloading or keeping the blockchain updated with a 24-thread CPU already takes up 10%-20% of the CPU, so it takes away performance for creating plots with CPU. Another option is that right now you can choose between a "wallet" mode and a "farming" mode, so it might be easier for the UI team to always leave the option to plot, so that if I only have the "wallet" option activated I can create plots without it affecting performance because the blockchain would not be constantly updating.

**Reply by OverActiveBladderSystem** [reply]
During the syncing process of the blockchain, high transaction volume, or large mempools the Full Node service can be an issue consuming cpu resources, but after fully sync'd the impact is much less, in the screen shot below I am running a fully sync'd Full Node, watching a video, writing this message, and a few other background tasks... the cpu spikes you see are "transaction blocks" which have the most impact but a 12600k (even with hyperthreading disabled) still manages to handle it with ease.

![Screenshot (1740)](https://github.com/Chia-Network/chia-blockchain-gui/assets/36348012/15341e9e-1a9d-4a31-9bd6-6dfa358732c5)

For the reasons I mentioned (high transaction periods, high mempool periods) it may be beneficial to move the Full Node service to a faster (single core speed) machine if you are not fully sync'd and in the process of plotting on the same machine (especially if you are making plots with the cpu).

Being able to utilize an existing Full Node already running on your local network makes sense (if even just to reduce overall bandwidth utilization from multiple Full Nodes running), having an easy way to do that in the GUI is something many windows users not comfortable with CLI would appreciate, so I definitely can see benefits in the ability to turn individual services on/off in the GUI, flexibility for each users unique situation is always a good thing.
